wasatch/FeatureIdentificationDevice.py

```diff
@@ -1304,243 +1441,184 @@ def set_tec_enable(self, flag):
             self.set_detector_tec_setpoint_degC(self.settings.eeprom.min_temp_degC)
 
         log.debug("Send detector TEC enable: %s", value)
-        ok = self.send_code(0xd6, value, label="SET_DETECTOR_TEC_ENABLE")
-        if ok:
+        ok = self._send_code(0xd6, value, label="SET_DETECTOR_TEC_ENABLE")
+        if ok.data:
             self.settings.state.tec_enabled = flag
         return ok
 
-    ##
-    # Set the source for incoming acquisition triggers.
-    #
-    # @param value either 0 for "internal" or 1 for "external"
-    #
-    # With internal triggering (the default), the spectrometer expects the
-    # USB host to explicitly send a START_ACQUISITION (ACQUIRE) opcode to
-    # begin each integration.  In external triggering, the spectrometer
-    # waits for the rising edge on a signal connected to a pin on the OEM
-    # accessory connector.
-    #
-    # Technically on ARM, the microcontroller is continuously monitoring
-    # both the external pin and listening for internal software opcodes.
-    # On the FX2 you need to explicitly place the microcontroller into
-    # external triggering mode to avail the feature.
-    def set_trigger_source(self, value):
+    def set_trigger_source(self, value: bool) -> SpectrometerResponse:
+        """
+        Set the source for incoming acquisition triggers.
+        @param value either 0 for "internal" or 1 for "external"
+        With internal triggering (the default), the spectrometer expects the
+        USB host to explicitly send a START_ACQUISITION (ACQUIRE) opcode to
+        begin each integration.  In external triggering, the spectrometer
+        waits for the rising edge on a signal connected to a pin on the OEM
+        accessory connector.
+        Technically on ARM, the microcontroller is continuously monitoring
+        both the external pin and listening for internal software opcodes.
+        On the FX2 you need to explicitly place the microcontroller into
+        external triggering mode to avail the feature.
+        """
         self.settings.state.trigger_source = value
         log.debug("trigger_source now %s", value)
 
         # Don't send the opcode on ARM. See issue #2 on WasatchUSB project
         if self.settings.is_arm():
-            return False
+            return SpectrometerResponse(data=False)
 
         msb = 0
         lsb = value
         buf = [0] * 8
 
         # MZ: this is weird...we're sending the buffer on an FX2-only command
```

Comment:
When we get a chance, we should re-test this on an FX2 with triggering, without sending the buffer.  This FW has been re-written and likely no longer requires this kludge.